### ISSUE TITLE
TypeResolution is enabled by default in designer, but exported rule does not enable it

### ISSUE THREAD

**Issue by aldaris** [pmd/pmd-designer#28]
**Describe the bug**
When using a rule that has pmd-java:typeIs checks, the rules are working fine in the designer, however the exported rules are not working later on when testing the rules in Java unit tests.
The rule would look something like:

```
//PrimaryPrefix/Name[@Image[ends-with(., 'Instant.now')]][pmd-java:typeIs('java.time.Instant')]
```

and the example code:
```
package foo;

public class Hello {

public static void main(String[] args) {
    java.time.Instant.now();
}
}
```

This file fails validation as expected in the designer, but when the rule is exported, the rule no longer works, and violation is no longer reported. After debugging TypeIsFunction, it turned out that this happened because the rule was not set up to perform typeResolution. Manually adding the `typeResolution="true"` attribute to the rule definition appears to fix this problem.

**To Reproduce**
1. Launch the designer from run.sh designer
1. Specify the above code snippet for the example
2. Use the rule listed above
3. There should be a matching node
4. Try to Export XPath to XML rule
5. Notice how the exported rule does not have typeResolution="true" flag

**Expected behavior**
Either the UI should run the queries with typeResolution disabled (and have a visible enable flag), or the export view should add the required attribute to the rule definition.

**Screenshots**
Let me know if you need one.

**Desktop (please complete the following information):**
 - OS: macOS 10.14.6
 - Java:
openjdk version "11.0.3" 2019-04-16
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.3+7)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.3+7, mixed mode)

**Comment by oowekyala** [comment]
Hmm, type resolution is enabled by default on Java rules (extending AbstractJavaRule), and we're going to remove the attribute for 7.0.0 anyway. So while we could technically make the designer add  `typeResolution='true'`, we have no way for now to find out whether the rule actually does use typeres...

I think the problem is that type resolution should also be enabled by default for Java XPath rules (or for all languages for that matter, Java being the only language supporting it).

**Comment by oowekyala** [comment]
Fixed by pmd/pmd#2083
